packages/core-worker/director.js

```diff
@@ -163,6 +163,8 @@ class WorkerDirector {
   static buildQueueSelector({
     status = [],
     selectTypes = [],
+    startDate,
+    endDate = new Date(),
```

Comment:
I'd go the other way round, default startDate to new Date(0) and then check if endDate is set or not, it makes it more clear for developers that the query will always fetch from the beginning if no explicit end or start date is set, an end date of new Date() could interfere with minor time differences, polling logic and such and throw up questions if that is safe to do so even though it is.